hera_cal/lstbin.py

```diff
@@ -0,0 +1,883 @@
+                  flag the entire bin.
+
+    atol : type=float, absolute tolerance for comparing LST bin center floats
+
+    median : type=boolean, if True use median for LST binning. Warning: this is slower.
+
+    truncate_empty : type=boolean, if True, truncate output time integrations that have no data
+                     in them.
+
+    sig_clip : type=boolean, if True, perform a sigma clipping algorithm of the LST bins on the
+               real and imag components separately. Warning: This is considerably slow.
+
+    sigma : type=float, input sigma threshold to use for sigma clipping algorithm.
+
+    min_N : type=int, minimum number of points in an LST bin to perform sigma clipping
+
+    return_no_avg : type=boolean, if True, return binned but un-averaged data and flags.
+
+    Output: (lst_bins, data_avg, flags_min, data_std, data_count)
+    -------
+    lst_bins : ndarray containing final lst grid of data
+
+    data_avg : dictionary of data having averaged in each LST bin
+
+    flags_min : dictionary of minimum of data flags in each LST bin
+    
+    data_std : dictionary of data with real component holding LST bin std along real axis
+               and imag component holding std along imag axis
+
+    data_count : dictionary containing the number count of data points averaged in each LST bin.
+
+    if return_no_avg:
+        Output: (data_bin, flags_min)
+        data_bin : dictionary with (ant1,ant2,pol) as keys and ndarrays holding
+            un-averaged complex visibilities in each LST bin as values. 
+        flags_min : dictionary with data flags
+    """
+    # get visibility shape
+    Ntimes, Nfreqs = data_list[0][data_list[0].keys()[0]].shape
+
+    # get dlst if not provided
+    if dlst is None:
+        dlst = np.median(np.diff(lst_list[0]))
+
+    # construct lst_grid
+    lst_grid = make_lst_grid(dlst, lst_start=lst_start, verbose=verbose)
+
+    # test for special case of lst grid restriction
+    if lst_low is not None and lst_hi is not None and lst_hi < lst_low:
+        lst_grid = lst_grid[(lst_grid > (lst_low - atol)) | (lst_grid < (lst_hi + atol))]
+    else:
+        # restrict lst_grid based on lst_low and lst_high
+        if lst_low is not None:
+            lst_grid = lst_grid[lst_grid > (lst_low - atol)]
+        if lst_hi is not None:
+            lst_grid = lst_grid[lst_grid < (lst_hi + atol)]
+
+    # Raise Exception if lst_grid is empty
+    if len(lst_grid) == 0:
+        raise ValueError("len(lst_grid) == 0; consider changing lst_low and/or lst_hi.")
+
+    # move lst_grid centers to the left
+    lst_grid_left = lst_grid - dlst / 2
+
+    # form new dictionaries
+    # data is a dictionary that will hold other dictionaries as values, which will
+    # themselves hold lists of ndarrays
+    data = odict()
+    flags = odict()
+    all_lst_indices = set()
+
+    # iterate over data_list
+    for i, d in enumerate(data_list):
```

Comment:
See if this for loop can be broken out into its own function for readability.

Reply:
turns out its not as easy as we thought, because this for loop is connected to the rest of the function via the parameter `d`, which this FOR loop (can) overwrite if a bl in conjugated, meaning if we separated this into its own function it would be reaching through namespaces to overwrite this variable, which seems like it could easily lead to more confusion...